packages/moonstone/Scrollable/ScrollableNative.js

```diff
@@ -744,49 +744,93 @@ class ScrollableBaseNative extends Component {
 		}
 	}
 
+	notifyVoiceException = (e, type) => {
+		const param = {'voiceUi': {'exception': type}};
+		if (window.webOSVoiceReportActionResult) {
+			window.webOSVoiceReportActionResult(param);
+			e.preventDefault();
+		}
+	}
+
+	canVoiceScroll = (e) => {
+		const
+			scroll = e && e.detail && e.detail.scroll,
+			{scrollTop, scrollLeft} = this.uiRef,
+			isRtl = this.uiRef.state.rtl;
+		const scrollBounds = this.uiRef.getScrollBounds();
+		const notifyAlreadyCompleted = () => this.notifyVoiceException(e, 'alreadyCompleted');
+
+		if (scroll === 'up' || scroll === 'top') {
+			if (scrollTop === 0) {
+				notifyAlreadyCompleted();
+				return false;
+			}
+		} else if (scroll === 'down' || scroll === 'bottom') {
+			if (scrollTop === scrollBounds.maxTop) {
```

Comment:
The condition might be `scrollTop >= scrollBounds.maxTop -1` because sometimes scrollTop is between scrollBounds.maxTop -1 and scrollBounds.maxTop when it reaches to the bottom in VirtualListNative or ScrollerNative (not VirtualList, Scroller).